lib/fetching/content-services/crowd-tangle-content-service.js

```diff
@@ -184,7 +185,28 @@ CrowdTangleContentService.prototype._parse = function(data, options) {
 	if (this.crowdtangle_lists[data.account.id]) {
 		metadata.ct_tag = this.crowdtangle_lists[data.account.id];
 	} else if (options.requestType == "SAVED_SEARCH") {
-		metadata.ct_tag = "Saved Search";
+		metadata.ct_tag = []
+		// This is a workaround to identify possible saved search names associated with a post using substring search.
+		// We plan to remove this workaround when CT adds list IDs/account IDs to saved search API responses.  
+		for(var saved_search_id in this.crowdtangle_saved_searches){
+			var saved_search = this.crowdtangle_saved_searches[saved_search_id];
+			try{
+				var regex = new RegExp(saved_search.words.join("|"), 'gi');
+			} catch (err) {
+				console.log("Error creating regex for saved searches");
+			}
+			if (saved_search.words.length && 
+				(regex.test(content)|| 
```

Comment:
Can you please test this with invalid content e.g. "foo(" and make sure it works as expected? (in that case it should fall back to generic "Saved search")

I think if we catch the error above, now `regex` will still be null and thus `regex.test` will throw.

Reply:
Fwiw, logic like this is often easier to deal with if you modularize it into a generic function. For example something like:

```js
const doesMatch = doesReportMatchWords(saved_search.words, content, metadata)
metadata.ct_tag.push(doesMatch ? saved_search.name : "Saved Search")
```

Reply:
Good catch. I moved the test statement inside the try block.

Reply:
And tested it with that example. It produces an error and then it will fall back to Saved Search.

Reply:
Yes. Modularized it. Moved the regex.test part inside the try block. "foo(" will throw an error and we would catch it. If no saved search matches are found, then we would add Saved Search default tag.